root/static/scripts/relationship-editor/components/RelationshipDialogContent.js

```diff
@@ -201,6 +205,57 @@ export function createInitialState(props: PropsT): RelationshipDialogStateT {
   };
 }
 
+/*
+ * Possibly updates newState.backward based on the source and target
+ * entities and the link type. The link direction is updated only in
+ * cases where the correct value can be inferred, e.g. "member of"
+ * relationships between person and group artist entities. See MBS-2604.
+ */
+function inferLinkDirection(newState: {...RelationshipDialogStateT}): void {
+  const source = newState.sourceEntity.source;
+  const target = newState.targetEntity.target;
+  const linkTypeId =
+    newState.linkType.autocomplete.selectedItem?.entity?.id ?? null;
+
+  if (
+    source.entityType === 'artist' &&
+    target.entityType === 'artist' &&
+    linkTypeId !== null &&
+    personGroupLinkTypeIds.has(linkTypeId)
+  ) {
+    const sourcePerson = source.typeID === ARTIST_TYPE_PERSON;
+    const sourceGroup =
+      source.typeID !== null && groupArtistTypeIds.has(source.typeID);
```

Comment:
I haven't used Flow before, but I didn't see any obvious way to do something similar to TypeScript's non-null assertion operator... hence all these null checks.

Reply:
It is probably safer to use our utility `nonEmpty`here.

Reply:
Sure, either way's fine with me -- `nonEmpty` has the advantage of being a bit shorter. :-)

I've made this change locally but I'll hold off on force-pushing since I assume it'll throw away all of these comments (sigh, GitHub).

Reply:
Sorry but after double-checking this change isn’t even needed.